### PR TITLE
prometheus best practices, TLS and basic auth support

### PR DESCRIPTION
**Pull request by yeoldegrove**
The main point was to enable TLS support.
One pre-requisite for this was a `log.Logger` object ([reference](https://pkg.go.dev/github.com/prometheus/exporter-toolkit@v0.7.1/web#ListenAndServe)) which is not available in the currently used logging package `logrus`.
I took this as an opportunity to port some best practices from e.g. [HAProxy exporter](https://github.com/prometheus/haproxy_exporter) and [MySQL Server exporter](https://github.com/prometheus/mysqld_exporter). This includes:
- using [promlog](https://pkg.go.dev/github.com/prometheus/common/promlog) instead of logrus for logging a38c72a2b6b9daf631e8dde7074e747a510c645d
  - DEPREACTION: log format changed due to this
- using [kingpin](https://pkg.go.dev/gopkg.in/alecthomas/kingpin.v2) instead if pflag to parse commandline a38c72a2b6b9daf631e8dde7074e747a510c645d
  - this is completely backwards compatible for the config file and command line argument setup  
    - some flags are marked as deprecated but will still work
  - includes a wrapper to parse `viper` config and overwrite it with `kinpin` flags
- add TLS and basic auth support by using [exporter-toolkit](https://pkg.go.dev/github.com/prometheus/exporter-toolkit)
- build binaries with `promu` 945cbc11d7c5093dcffa02f46b751d8ee1f0b683
- test: build, run and test binary 3017a2fa77ccf61bcbdc5b26b501e5b9bd374e35

In addition to that, the OBS + release CI is kind of broken at the moment. These commits tend to fix it:
- fix github url in specfile e219a0222e5b0ff7f7ecc0cf3692d74170a4fd08
- fix github release e08c01af4c2a7c5436ebf84b16ed50f2829f9e84

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,47 +30,65 @@ ARCHS ?= amd64 arm64 ppc64le s390x
 
 default: clean mod-tidy generate fmt vet-check test build
 
-build: amd64
+promu-prepare: 
+	sed "s/{{.Version}}/$(VERSION)/" .promu.yml >.promu.release.yml
```
Comment (stefanotorresi) [inline Makefile:34]: I guess we should to add `.promu.release.yml` to .gitignore now, right?

File: ha_cluster_exporter.yaml [hunk 2]
```diff
@@ -1,12 +1,16 @@
 # sample config
-port: "9664"
-address: "0.0.0.0"
-log-level: "info"
+web:
+  listen-address: "0.0.0.0:9664"
+  telemetry-path: "/metrics"
+  config:
+    file: "/etc/ha_cluster_exporter.web.yaml"
```
Comment (stefanotorresi) [inline ha_cluster_exporter.yaml:6]: hmm, where is this file coming from?

**Review by stefanotorresi — COMMENTED** [review]
Some questions inline just to remind myself, let's have a chat face to face to discuss some of the more extensive changes.

File: .promu.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+repository:
+    path: github.com/ClusterLabs/ha_cluster_exporter
+build:
+    flags: -a -tags netgo
+    ldflags: |
+        -X github.com/prometheus/common/version.Version={{.Version}}
+        -X github.com/prometheus/common/version.Revision={{.Revision}}
+        -X github.com/prometheus/common/version.Branch={{.Branch}}
+        -X github.com/prometheus/common/version.BuildUser={{user}}@{{host}}
+        -X github.com/prometheus/common/version.BuildDate={{date "20060102-15:04:05"}}
```
[the remark this answers is not in the document]
Comment (yeoldegrove) [reply .promu.yml:13]: This is snatched from https://github.com/prometheus/haproxy_exporter/blob/b94dc26d11b48e2c447022abd78aa00776dc4ed3/.promu.yml#L14 and https://github.com/prometheus/mysqld_exporter/blob/3d2a10b014e2268890d0aff57de319b6ff78ddd7/.promu.yml#L14 and is later set to the actual build date by `promu`.

File: ha_cluster_exporter.yaml (hunk 2 above)
Comment (yeoldegrove) [reply ha_cluster_exporter.yaml:6]: The format is defined in the README.
If it is not existing the exporter will basically start without it (TLS/basic-auth disabled).

File: Makefile (hunk 1 above)
Comment (yeoldegrove) [reply Makefile:34]: fixed in force push

**Comment by yeoldegrove** [comment]
@stefanotorresi force pushed the changes you wanted. It is ready to merged from my point of view.

**Review by stefanotorresi — APPROVED** [review]

**Pull request merged**